### PR TITLE
Update build-plan CHANGELOG.md

### PR DESCRIPTION
**Pull request by mpeddada1**

### CODE REVIEW

**Comment by loosebazooka** [comment]
@chanseokoh have we done a release with any of these changes before? I don't see any versioning in here?

**Comment by chanseokoh** [comment]
Ah, yeah, sorry. I forgot to update the template. The current `[unreleased]` entries need to be under 0.3.1. (Strictly speaking, we should have both 0.3.0 and 0.3.1, but I think it's fine to go with a single 0.3.1.)

**Comment by chanseokoh** [comment]
Good catch!

File: jib-build-plan/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ All notable changes to this project will be documented in this file.
 ### Changed
 
 - Removed `get/setOsHint()` and `get/setArchitectureHint()` in favor of `get/setPlatforms()` and `addPlatform()`. ([#2584](https://github.com/GoogleContainerTools/jib/pull/2584))
+- Replaced `BiFunction` usage in `FileEntriesLayer` with `FilePermissionsProvider`, `ModificationTimeProvider` and `OwnershipProvider`. ([#2643](https://github.com/GoogleContainerTools/jib/pull/2643))
```
Comment (chanseokoh) [inline jib-build-plan/CHANGELOG.md:14]: I think linking #2638 is better for the user looking for more information. #2643 doesn't have any description.

File: jib-build-plan/CHANGELOG.md [hunk 2]
```diff
@@ -5,12 +5,21 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+
 - Added `Platform` class representing an image platform. ([#2584](https://github.com/GoogleContainerTools/jib/pull/2584))
 - Added `get/setPlatforms()` and `addPlatform()` to `ContainerBuildPlan` for setting and getting image platforms. ([#2584](https://github.com/GoogleContainerTools/jib/pull/2584))
 
 ### Changed
 
 - Removed `get/setOsHint()` and `get/setArchitectureHint()` in favor of `get/setPlatforms()` and `addPlatform()`. ([#2584](https://github.com/GoogleContainerTools/jib/pull/2584))
+- Replaced `BiFunction` usage in `FileEntriesLayer` with `FilePermissionsProvider`, `ModificationTimeProvider` and `OwnershipProvider`. ([#2638](https://github.com/GoogleContainerTools/jib/issues/2638))
```
Comment (chanseokoh) [inline jib-build-plan/CHANGELOG.md:22]: Now this one should go under `[unreleased]`. 0.3.1 is already released and live.

Comment (mpeddada1) [reply jib-build-plan/CHANGELOG.md]: Oh right. Sorry about that!

**Comment by loosebazooka** [comment]
Yeah we probably should've provided some better guidance on this file. Anyway, while you're in here can you update the top to look more like https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin/CHANGELOG.md#change-log including a title and `[unreleased]` header and stuff

**Comment by loosebazooka** [comment]
I think we still need the "changelog" title and subtitle, and the "unreleased" subsection header?

**Pull request merged**